### DISCUSSION TITLE
NodeJS affected by Chromium bug?

### DISCUSSION THREAD

**Discussion by hello-smile6** [nodejs/help#4876] [Q&A] [answered]
In Chromium, the following code causes a crash. Is Node affected?
`alert(JSON.stringify(new Array(90000000).fill(0)))`

**Comment by VoltrexKeyva** [comment, marked as answer]
This is possibly a V8 limitation or the runtime simply runs out of memory required enough to allocate to create an array with 90 million elements referencing to each other as filled by `Array.prototype.fill()`. Can't really call this a bug.

**Reply by hello-smile6** [reply]
How much RAM does the array use as a string?

**Reply by VoltrexKeyva** [reply]
The stringified array doesn't really matter in this case, as doing the same thing without stringifying causes the program to crash if there isn't enough memory to allocate. That being said, the amount of memory the construction of an array allocates depends on the amount of elements it holds; in your case, you're creating an array with 90 million elements, all referencing each other as they are filled with the built-in `Array.prototype.fill()` method.

**Comment by jasnell** [comment]
Also, we don't implement `alert()`. Some systems with limited memory will have a problem with this and likely crash with an out of memory error. Just tested with node 16 on my dev machine and nothing crashes.

**Reply by hello-smile6** [reply]
Use console.log() instead.
